apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/newchat/NewChatView.kt

```diff
@@ -0,0 +1,435 @@
+}
+
+@Composable
+private fun RetryButton(onClick: () -> Unit) {
+  Column(
+    Modifier.fillMaxSize(),
+    horizontalAlignment = Alignment.CenterHorizontally,
+    verticalArrangement = Arrangement.Center
+  ) {
+    IconButton(onClick, Modifier.size(30.dp)) {
+      Icon(painterResource(MR.images.ic_refresh), null)
+    }
+    Spacer(Modifier.height(DEFAULT_PADDING))
+    Text(stringResource(MR.strings.retry_verb))
+  }
+}
+
+@Composable
+private fun InviteView(rhId: Long?, connReqInvitation: String, contactConnection: MutableState<PendingContactConnection?>) {
+  SectionView(stringResource(MR.strings.share_this_1_time_link).uppercase()) {
+    LinkTextView(connReqInvitation, true)
+  }
+
+  Spacer(Modifier.height(10.dp))
+
+  SectionView(stringResource(MR.strings.or_show_this_qr_code).uppercase()) {
+    SimpleXLinkQRCode(connReqInvitation, onShare = { chatModel.markShowingInvitationUsed() })
+  }
+
+  Spacer(Modifier.height(10.dp))
+  val incognito = remember { mutableStateOf(controller.appPrefs.incognito.get()) }
+  IncognitoToggle(controller.appPrefs.incognito, incognito) {
+    if (appPlatform.isDesktop) ModalManager.end.closeModals()
+    ModalManager.end.showModal { IncognitoView() }
+  }
+  KeyChangeEffect(incognito.value) {
+    withBGApi {
+      val contactConn = contactConnection.value ?: return@withBGApi
+      val conn = controller.apiSetConnectionIncognito(rhId, contactConn.pccConnId, incognito.value) ?: return@withBGApi
+      contactConnection.value = conn
+      chatModel.updateContactConnection(rhId, conn)
+    }
+    chatModel.markShowingInvitationUsed()
+  }
+  SectionTextFooter(sharedProfileInfo(chatModel, incognito.value))
+}
+
+@Composable
+private fun AddContactLearnMoreButton() {
+  IconButton(
+    {
+      if (appPlatform.isDesktop) ModalManager.end.closeModals()
+      ModalManager.end.showModalCloseable { close ->
+        Column(
+          Modifier
+            .fillMaxHeight()
+            .padding(horizontal = DEFAULT_PADDING),
+          verticalArrangement = Arrangement.SpaceBetween
+        ) {
+          AddContactLearnMore(close)
+        }
+      }
+    }
+  ) {
+    Icon(
+      painterResource(MR.images.ic_info),
+      stringResource(MR.strings.learn_more),
+    )
+  }
+}
+
+@Composable
+private fun ConnectView(rhId: Long?, showQRCodeScanner: MutableState<Boolean>, pastedLink: MutableState<String>, close: () -> Unit) {
+  SectionView(stringResource(MR.strings.paste_the_link_you_received).uppercase()) {
+    PasteLinkView(rhId, pastedLink, showQRCodeScanner, close)
+  }
+
+  if (appPlatform.isAndroid) {
+    Spacer(Modifier.height(10.dp))
+
+    SectionView(stringResource(MR.strings.or_scan_qr_code).uppercase()) {
+      QRCodeScanner(showQRCodeScanner) { text ->
+        withBGApi {
+          val res = verify(rhId, text, close)
+          if (!res) {
+            AlertManager.shared.showAlertMsg(
+              title = generalGetString(MR.strings.incorrect_code)
```

Comment:
reuses key from VerifyCodeView, should say something about link instead, e.g.
title: "Invalid QR code", message: "The code you scanned is not a SimpleX link QR code."